blog/_posts/2022-03-15-pulumi.md

```diff
@@ -0,0 +1,275 @@
+  trying to deploy at the same time.
+  
+This blog post is a quick guide to how to set up Pulumi for local development,
+so that you can iterate quickly without messing with YAML.
+
+## What Converting to Pulumi Looks Like
+
+This is the Tilt blog. So this is eventually going to lead up to the [Tilt
+Pulumi
+extension](https://github.com/tilt-dev/tilt-extensions/blob/master/pulumi/README.md).
+
+But first let's look at an app that uses Pulumi to define Deployments.
+
+Here's what a normal Deployment looks like:
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: helloworld
+spec:
+  selector:
+    matchLabels:
+      app: helloworld
+  template:
+    metadata:
+      labels:
+        app: helloworld
+    spec:
+      containers:
+      - name: helloworld
+        image: helloworld-image
+```
+
+Pulumi's APIs try to closely mirror what this YAML looks like, providing exactly
+the same arguments. Here's an example that uses Javascript (though Pulumi also
+has APIs for Python, Typescript, and Go):
+
+```javascript
+"use strict";
+const k8s = require("@pulumi/kubernetes");
+
+const deployment = new k8s.apps.v1.Deployment("pulumi-helloworld", {
+  spec: {
+    selector: { 
+      matchLabels: { app: "pulumi-helloworld" } 
+    },
+    replicas: 1,
+    template: {
+      metadata: { labels: { app: "pulumi-helloworld" } },
+      spec: { 
+        containers: [{ 
+          name: "pulumi-helloworld", 
+          image: "pulumi-helloworld-image" 
+        }] 
+      }
+    }
+  }
+});
+exports.name = deployment.metadata.name;
+```
+
+And because this is a real programming language, it's easy to factor out the
+common constants.
+
+```javascript
+"use strict";
+const k8s = require("@pulumi/kubernetes");
+
+const appLabels = { app: "pulumi-helloworld" };
+const deployment = new k8s.apps.v1.Deployment("pulumi-helloworld", {
+  spec: {
+    selector: { 
+      matchLabels: appLabels 
+    },
+    replicas: 1,
+    template: {
+      metadata: { labels: appLabels },
+      spec: { 
+        containers: [{ 
+          name: "pulumi-helloworld", 
+          image: "pulumi-helloworld-image" 
+        }] 
+      }
+    }
+  }
+});
+exports.name = deployment.metadata.name;
+```
+
+## Setting up a Fast Local Dev Loop
+
+I've built a lot of developer tools! And I started to notice how every tool
+would gradually expand in scope to add the same set of features to make them
+iterating on them easy:
+
+- File watches and dependency tracking for auto-builds.
+
+- Terminal & UI dashboards for real-time status.
+
+- Diagnostic CLIs for inspecting when things went wrong.
+
+We built Tilt to be a dev environment that integrates with any build/deploy tool.
+That's why Tilt has a pluggable system for deploying to Kubernetes, such that we
+can layer on our own image builds and live updates for local development.
+
+Let's take a look at how it works!
+
+First, we modify our Pulumi script to take an image as an argument:
```

Comment:
Having not used Pulumi before (and briefly skimming over the JS), I assumed this meant CLI arg and was a little confused by what magic was going on with `image_configs` in the `pulumi_resource`.

Maybe something along the lines of "read the image name from config (that we'll generate and pass from the Tiltfile)" but that's really verbose